src/quacc/util/defects.py

```diff
@@ -0,0 +1,174 @@
+    charge_state: int = 0,
+    sc_mat: np.ndarray | None = None,
+    min_atoms: int = 80,
+    max_atoms: int = 240,
+    min_length: float = 10.0,
+    force_diagonal: bool = False,
+    **defectgen_kwargs,
+) -> list[Atoms]:
+    """
+    Function to make defects from a bulk atoms object.
+
+    Parameters
+    ----------
+    atoms
+        bulk atoms
+    defectgen
+        defect generator
+    sc_mat
+        supercell matrix
+    min_atoms
+        minimum number of atoms in supercell
+    max_atoms
+        maximum number of atoms in supercell
+    min_length
+        minimum length of supercell
+    force_diagonal
+        force supercell to be diagonal
+    charge_state
+        charge state of defect
+    **defectgen_kwargs
+        keyword arguments to pass to the pymatgen.analysis.defects.generators get_defects() method
+
+    Returns
+    -------
+    list[Atoms]
+        All generated defects
+    """
+
+    # Use pymatgen-analysis-defects and ShakeNBreak to generate defects
+    struct = AseAtomsAdaptor.get_structure(atoms)
+    atoms_info = atoms.info.copy()
+
+    # Make all the defects
+    defects = defectgen().get_defects(struct, **defectgen_kwargs)
+    final_defects = []
+    for defect in defects:
+        defect.user_charges = [charge_state]
+
+        # Generate the supercell for a defect
+        defect_supercell = defect.get_supercell_structure(
+            sc_mat=sc_mat,
+            dummy_species=DummySpecies(),
+            min_atoms=min_atoms,
+            max_atoms=max_atoms,
+            min_length=min_length,
+            force_diagonal=force_diagonal,
+        )
+
+        # Generate DefectEntry object from Defect object
+        defect_entry = _get_defect_entry_from_defect(
+            defect=defect, defect_supercell=defect_supercell, charge_state=charge_state
+        )
+
+        # Instantiate class to apply rattle and bond distortion to all defects
+        Dist = Distortions(defects=[defect_entry])
+
+        # Apply rattle and bond distortion to all defects
+        defect_dict, distortion_metadata = Dist.apply_distortions()
+        defect_symbol = list(distortion_metadata["defects"].keys())[0]
+        distortion_dict = defect_dict[defect_symbol]["charges"][charge_state][
+            "structures"
+        ]["distortions"]
+
+        # Make atoms objects and store defect stats
+        for distortions, defect_struct in distortion_dict.items():
+            final_defect = AseAtomsAdaptor.get_atoms(defect_struct)
+            final_defect.info = atoms_info.copy()
+            defect_stats = {
+                "defect_symbol": defect_symbol,
+                "charge_state": charge_state,
+                "distortions": distortions,
+                "bulk": atoms,
+                "defect": defect,
+            }
+            final_defect.info["defect_stats"] = defect_stats
+            final_defects.append(final_defect)
+    return final_defects
+
+
+def _get_defect_entry_from_defect(
```

Comment:
I renamed this with a `_` prefix to indicate that this isn't really a user-facing function.